core/typedefs.h

```diff
@@ -71,6 +71,31 @@
 #endif
 #endif
 
+// No discard allows the compiler to flag warnings if we don't use the return value of functions / classes
+#ifndef _NO_DISCARD_
```

Comment:
The macro will be good for `3.x`, but for `master` we should just use `[[nodiscard]]` as the fallback will never be needed.